### PR TITLE
Fixed upside-down rendering of top-down view

### PR DESCRIPTION
**Pull request by neworderofjamie**
I meant to fix this in #185 but totally forgot to. The basic problem was an OpenGL one but there were additional issues with correctly rendering routes and vector fields **above** the ground. This also fixes issues with the ant world map creator and takes advantage of ``Vector2<>`` and ``Pose2<>`` for various bits of route API

### CODE REVIEW

File: src/antworld/renderer.cc [hunk 1]
```diff
@@ -232,18 +232,19 @@ void Renderer::renderTopDownView(GLint viewportX, GLint viewportY, GLsizei viewp
...
             minBound[1].value(), maxBound[1].value(),
-            -1.0, maxBound[2].value() - minBound[2].value());
+            0.0, maxBound[2].value() - minBound[2].value());
 
-    // Build modelview matrix to centre world
+    // Build modelview matrix so that the 'top' is up against the near clip plane
     glMatrixMode(GL_MODELVIEW);
     glLoadIdentity();
-    glScalef(1.0f, 1.0f, -1.0f);
+    glTranslatef(0.0f, 0.0f, -maxBound[2].value());
```
Comment (alexdewar) [inline src/antworld/renderer.cc:247]: This could use a ``static_cast<GLfloat>(...)`` for extra fussy compilers...

Comment (neworderofjamie) [reply src/antworld/renderer.cc:247]: oh yes - missed one

File: src/antworld/route_continuous.cc [hunk 2]
```diff
@@ -246,35 +246,35 @@ void RouteContinuous::render(meter_t antX, meter_t antY, degree_t antHeading) co
...
+bool RouteContinuous::atDestination(const Vector2<meter_t> &position, meter_t threshold) const
 {
     // If route's empty, there is no destination so return false
     if(m_Waypoints.empty()) {
         return false;
     }
     // Otherwise return true if
     else {
-        return distance(m_Waypoints.back(), x, y) < threshold;
+        return distance(m_Waypoints.back(), position.x(), position.y()) < threshold;
```
Comment (alexdewar) [inline src/antworld/route_continuous.cc:266]: You could also do ``m_Waypoints.back().distance2D(position)`` instead.

Comment (neworderofjamie) [reply src/antworld/route_continuous.cc:266]: Sadly that doesn't work as ``m_Waypoints`` are passed directly to OpenGL so are unitless ``GLfloat``s

**Pull request merged**